BestFloww/BestFlow
# Base button icons

## Pull Request Summary
Added new Icon component that renders an svg icon with some additional customization based on props. The icon's name is passed as props, then matched to the appropriate svg in Icons.svg. Additional props can be passed to (slightly) customize the Icon.

Icon can be used as a standalone component, but is also a child component of BaseButton. If desired, icon-related props can be passed to BaseButton, which will then be passed to Icon and properly rendered (there are tests for this in BaseButton.test, beyond tests for the Icon in general in Icon.test).

A new package "jest-svg-transformer" was added to properly test components that import svg files.

Implementation based on: https://stackoverflow.com/questions/42296499/how-to-display-svg-icons-svg-files-in-ui-using-react-component.

### Change types


### Testing

### How to manually test this PR (QA instructions)
![image](https://user-images.githubusercontent.com/69657338/197834286-12bb1849-a16c-4055-94fc-1e7e40986d5b.png)
Add this tag (or similar) to a button of your choice and see the change.

## Questions
